src/realm/obj.cpp

```diff
@@ -661,6 +661,8 @@ ObjKey Obj::get_backlink(ColKey backlink_col, size_t backlink_ndx) const
 
 std::vector<ObjKey> Obj::get_all_backlinks(ColKey backlink_col) const
 {
+    update_if_needed();
```

Comment:
I think this is correct, but I would really like to know if it was visible and could fix an issue we've been seeing. Are you able to write a test case for it?

Reply:
I don't think it is possible to make a test where this addition matters. This change is related to some modifications on TableView I have in the pipeline where it actually matters. Perhaps I should have postponed this change until then, but I wanted to separate out all changes that were not directly related to the changes to TableView.

Reply:
The reason it does not matter today is that this function is currently only called on a fresh object.

Reply:
Seems useful to leave out any functional changes together with when it makes sense.